src/device/pf_block_writer.c

```diff
@@ -4109,3 +4128,441 @@ void pf_put_pdport_data_adj (
+      is_big_endian,
+      &p_tower->floor_4,
+      res_len,
+      p_bytes,
+      p_pos);
+
+   pf_put_rpc_floor_5_ip (
+      is_big_endian,
+      &p_tower->floor_5,
+      res_len,
+      p_bytes,
+      p_pos);
+}
+
+/**
+ * Put rpc handle into a buffer
+ * @param is_big_endian    In:   Endianness of the destination buffer.
+ * @param p_handle         In:   Rpc handle
+ * @param res_len          In:   Size of destination buffer.
+ * @param p_bytes          Out:  Destination buffer.
+ * @param p_pos            InOut:Position in destination buffer.
+ */
+static void pf_put_rpc_handle (
+   bool is_big_endian,
+   const pf_rpc_handle_t * p_handle,
+   uint16_t res_len,
+   uint8_t * p_bytes,
+   uint16_t * p_pos)
+{
+   pf_put_uint32 (
+      is_big_endian,
+      p_handle->rpc_entry_handle,
+      res_len,
+      p_bytes,
+      p_pos);
+
+   pf_put_rpc_uuid (
+      is_big_endian,
+      &p_handle->handle_uuid,
+      res_len,
+      p_bytes,
+      p_pos);
+}
+
+/**
+ * Put rpc epm entry into a buffer. If no no protocol tower is
+ * configured / actual_count == 0, a empty entry is written.
+ * @param is_big_endian    In:   Endianness of the destination buffer.
+ * @param p_entry          In:   Rpc epm entry including protocol
+ *                               tower configuration.
+ * @param res_len          In:   Size of destination buffer.
+ * @param p_bytes          Out:  Destination buffer.
+ * @param p_pos            InOut:Position in destination buffer.
+ */
+static void pf_put_rpc_epm_entry (
+   bool is_big_endian,
+   const pf_rpc_entry_t * p_entry,
+   uint16_t res_len,
+   uint8_t * p_bytes,
+   uint16_t * p_pos)
+{
+   pf_put_uint32 (is_big_endian, p_entry->max_count, res_len, p_bytes, p_pos);
+
+   pf_put_uint32 (is_big_endian, p_entry->offset, res_len, p_bytes, p_pos);
+
+   pf_put_uint32 (is_big_endian, p_entry->actual_count, res_len, p_bytes, p_pos);
+
+   /*Add entry if there if not empty */
+   if (p_entry->actual_count > 0)
+   {
+      pf_put_rpc_uuid (
+         is_big_endian,
+         &p_entry->object_uuid,
+         res_len,
+         p_bytes,
+         p_pos);
+
+      pf_put_tower_entry (
+         is_big_endian,
+         &p_entry->tower_entry,
+         res_len,
+         p_bytes,
+         p_pos);
+   }
+}
+
+void pf_put_lookup_response_data (
+   pnet_t * net,
+   bool is_big_endian,
+   const pf_rpc_lookup_rsp_t * p_lookup_rsp,
+   uint16_t res_len,
+   uint8_t * p_bytes,
+   uint16_t * p_pos)
+{
+   uint8_t pad8 = 0;
+
+   pf_put_rpc_handle (
+      is_big_endian,
+      &p_lookup_rsp->rpc_handle,
+      res_len,
+      p_bytes,
+      p_pos);
+
+   pf_put_uint32 (
+      is_big_endian,
+      p_lookup_rsp->num_entry,
+      res_len,
+      p_bytes,
+      p_pos);
+
+   pf_put_rpc_epm_entry (
+      is_big_endian,
+      &p_lookup_rsp->rpc_entry,
+      res_len,
+      p_bytes,
+      p_pos);
+
+   /*Add padding if needed*/
+   if (*p_pos % 2 > 0)
+   {
+      pf_put_byte (pad8, res_len, p_bytes, p_pos);
```

Comment:
Maybe just `pf_put_byte (0, ...`  
I think that is more obvious to the reader.

Reply:
done